lib/pbench/server/api/resources/query_apis/datasets/__init__.py

```diff
@@ -79,34 +87,40 @@ class IndexMapBase(ElasticBase):
         "contents": {"index": "run-toc", "whitelist": ["directory", "files"]},
     }
 
-    def __init__(self, config: PbenchServerConfig, logger: Logger, schema: Schema):
-        if "name" not in schema:
-            raise MissingDatasetNameParameter(self.__class__.__name__)
-        super().__init__(config, logger, schema)
-
-    def preprocess(self, client_json: JSON) -> CONTEXT:
+    def __init__(self, config: PbenchServerConfig, logger: Logger, *schemas: ApiSchema):
+        super().__init__(config, logger, *schemas)
+
+        api_name = self.__class__.__name__
+
+        if not schemas:
+            raise MissingDatasetNameParameter(api_name, "no schema provided")
+        dset = self.schemas.get_param_by_type(
+            API_METHOD.POST,
+            ParamType.DATASET,
+            ApiParams(body=None, query=None, uri=None),
```

Comment:
I believe that you can set up the fields of `ApiParams` with default values, in which case we could reduce this line to just
```
            ApiParams(),
```
which would be nicer.

Reply:
It's not a common pattern; is that really worth another commit?

Reply:
Not on its own, no.  But, there might be other things which justify another commit....